frontend/public/components/machine-autoscaler.tsx

```diff
@@ -0,0 +1,128 @@
+/* eslint-disable no-undef, no-unused-vars */
+import * as React from 'react';
+import * as _ from 'lodash-es';
+
+import { MachineAutoscalerModel } from '../models';
+import { groupVersionFor, K8sResourceKind, referenceForGroupVersionKind, referenceForModel } from '../module/k8s';
+import { ColHead, DetailsPage, List, ListHeader, ListPage } from './factory';
+import {
+  Kebab,
+  navFactory,
+  ResourceKebab,
+  ResourceLink,
+  ResourceSummary,
+  SectionHeading,
+} from './utils';
+
+const { common } = Kebab.factory;
+const menuActions = [...common];
+const machineAutoscalerReference = referenceForModel(MachineAutoscalerModel);
+
+const MachineAutoscalerTargetLink: React.FC<MachineAutoscalerTargetLinkProps> = ({obj}) => {
+  const targetAPIVersion: string = _.get(obj, 'spec.scaleTargetRef.apiVersion');
+  const targetKind: string = _.get(obj, 'spec.scaleTargetRef.kind');
+  const targetName: string = _.get(obj, 'spec.scaleTargetRef.name');
+  if (!targetAPIVersion || !targetKind || !targetName) {
+    return <React.Fragment>-</React.Fragment>;
```

Comment:
Can this just be:

```jsx
return '-';
```

Reply:
TypeScript doesn't like it :(

Reply:
Hmm, okay. 🤷‍♂

Reply:
Yeah, I think it's a problem with the React type definitions, but `React.Fragment` works

```
ERROR in /Users/sam/go/src/github.com/openshift/console/frontend/public/components/machine-autoscaler.tsx(21,7):
TS2322: Type '({ obj }: PropsWithChildren<MachineAutoscalerTargetLinkProps>) => Element | "-"' is not assignable to type 'FunctionComponent<MachineAutoscalerTargetLinkProps>'.
  Type 'Element | "-"' is not assignable to type 'ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)>) | (new (props: any) => Component<any, any, any>)>'.
    Type '"-"' is not assignable to type 'ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)>) | (new (props: any) => Component<any, any, any>)>'.
```

Reply:
Seems to be a known issue. https://github.com/DefinitelyTyped/DefinitelyTyped/issues/20544